diff --git a/node_modules/react-native/ReactAndroid/build.gradle b/node_modules/react-native/ReactAndroid/build.gradle
index 0859fcd..27acebd 100644
--- a/node_modules/react-native/ReactAndroid/build.gradle
+++ b/node_modules/react-native/ReactAndroid/build.gradle
@@ -89,7 +89,12 @@ task prepareFolly(dependsOn: dependenciesPath ? [] : [downloadFolly], type: Copy
 }

 task prepareHermes() {
-    def hermesAAR = new File("$projectDir/../node_modules/hermesvm/android/hermes-debug.aar")
+    def hermesAAR = file("$projectDir/../node_modules/hermesvm/android/hermes-debug.aar")
+    if (!hermesAAR.exists()) {
+        // For an app to build from RN source, hermesvm is located at /path/to/app/node_modules
+        // and $projectDir is located at /path/to/app/node_modules/react-native/ReactAndroid
+        hermesAAR = file("$projectDir/../../hermesvm/android/hermes-debug.aar")
+    }
     def soFiles = zipTree(hermesAAR).matching({ it.include "**/*.so" })

     copy {
diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/hermes/reactexecutor/Android.mk b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/hermes/reactexecutor/Android.mk
index adca702..54e19f6 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/hermes/reactexecutor/Android.mk
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/hermes/reactexecutor/Android.mk
@@ -12,7 +12,7 @@ LOCAL_MODULE := hermes-executor-release

 LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp)

-LOCAL_C_INCLUDES := $(LOCAL_PATH) $(REACT_NATIVE)/ReactCommon/jsi $(REACT_NATIVE)/node_modules/hermesvm/android/include
+LOCAL_C_INCLUDES := $(LOCAL_PATH) $(REACT_NATIVE)/ReactCommon/jsi $(REACT_NATIVE)/node_modules/hermesvm/android/include $(REACT_NATIVE)/../hermesvm/android/include

 LOCAL_CPP_FEATURES := exceptions

@@ -30,7 +30,7 @@ LOCAL_CFLAGS := -DHERMES_ENABLE_DEBUGGER=1

 LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp)

-LOCAL_C_INCLUDES := $(LOCAL_PATH) $(REACT_NATIVE)/ReactCommon/jsi $(REACT_NATIVE)/node_modules/hermesvm/android/include
+LOCAL_C_INCLUDES := $(LOCAL_PATH) $(REACT_NATIVE)/ReactCommon/jsi $(REACT_NATIVE)/node_modules/hermesvm/android/include $(REACT_NATIVE)/../hermesvm/android/include

 LOCAL_CPP_FEATURES := exceptions

diff --git a/node_modules/react-native/ReactCommon/hermes/inspector/Android.mk b/node_modules/react-native/ReactCommon/hermes/inspector/Android.mk
index 46eaede..00cb583 100644
--- a/node_modules/react-native/ReactCommon/hermes/inspector/Android.mk
+++ b/node_modules/react-native/ReactCommon/hermes/inspector/Android.mk
@@ -15,7 +15,7 @@ LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp $(LOCAL_PATH)/detail/*.cpp $(L
 LOCAL_C_ROOT := $(LOCAL_PATH)/../..

 LOCAL_CFLAGS := -DHERMES_ENABLE_DEBUGGER=1
-LOCAL_C_INCLUDES := $(LOCAL_C_ROOT) $(REACT_NATIVE)/ReactCommon/jsi $(REACT_NATIVE)/node_modules/hermesvm/android/include
+LOCAL_C_INCLUDES := $(LOCAL_C_ROOT) $(REACT_NATIVE)/ReactCommon/jsi $(REACT_NATIVE)/node_modules/hermesvm/android/include $(REACT_NATIVE)/../hermesvm/android/include
 LOCAL_EXPORT_C_INCLUDES := $(LOCAL_C_ROOT)

 LOCAL_CPP_FEATURES := exceptions
